test/it/controller/etcdopstask/reconciler_test.go

```diff
@@ -0,0 +1,358 @@
+
+// testEtcdOpsTaskAddFinalizer tests that the finalizer is added to the EtcdOpsTask upon creation.
+func testEtcdOpsTaskAddFinalizer(t *testing.T, namespace string, reconcilerTestEnv ReconcilerTestEnv) {
+	g := NewWithT(t)
+	etcdopstaskInstance := newTestTask(namespace, nil)
+	cl := reconcilerTestEnv.itTestEnv.GetClient()
+	ctx := context.Background()
+	g.Expect(cl.Create(ctx, etcdopstaskInstance)).To(Succeed())
+
+	g.Eventually(func() bool {
+		if err := cl.Get(ctx, client.ObjectKeyFromObject(etcdopstaskInstance), etcdopstaskInstance); err != nil {
+			return false
+		}
+		return slices.Contains(etcdopstaskInstance.Finalizers, druidapicommon.EtcdOpsTaskFinalizerName)
+	}, 5*time.Second, 100*time.Millisecond).Should(BeTrue(), "finalizer should be added to etcdopstask")
+
+	t.Logf("test completed: finalizer %s was successfully added to etcdopstask %s/%s", druidapicommon.EtcdOpsTaskFinalizerName, etcdopstaskInstance.Namespace, etcdopstaskInstance.Name)
+}
+
+// testEtcdOpsTaskRejctedIfEtcdBackupDisabled tests that an EtcdOpsTask is rejected if the referenced etcd has backup disabled.
+func testEtcdOpsTaskRejectedIfEtcdBackupDisabled(t *testing.T, namespace string, reconcilerTestEnv ReconcilerTestEnv) {
+	g := NewWithT(t)
+
+	etcdInstance := testutils.EtcdBuilderWithDefaults("test-etcd", namespace).Build()
+	etcdInstance.Spec.Backup = druidv1alpha1.BackupSpec{}
+
+	cl := reconcilerTestEnv.itTestEnv.GetClient()
+	ctx := context.Background()
+
+	g.Expect(cl.Create(ctx, etcdInstance)).To(Succeed())
+	t.Logf("created etcd instance %s/%s with backup disabled", etcdInstance.Namespace, etcdInstance.Name)
+
+	g.Expect(etcdInstance.IsBackupStoreEnabled()).To(BeFalse(), "backup should be disabled for this test")
+
+	etcdOpsTaskInstance := newTestTask(namespace, nil)
+	g.Expect(cl.Create(ctx, etcdOpsTaskInstance)).To(Succeed())
+	t.Logf("created etcdopstask %s/%s", etcdOpsTaskInstance.Namespace, etcdOpsTaskInstance.Name)
+
+	expectedErrorCode := ondemandsnapshot.ErrBackupNotEnabled
+	assertEtcdOpsTaskStateAndErrorCode(ctx, g, t, cl, etcdOpsTaskInstance, druidv1alpha1.TaskStateRejected, druidv1alpha1.LastOperationStateError, druidv1alpha1.OperationPhaseAdmit, &expectedErrorCode)
+
+	t.Logf("test completed: etcdopstask was correctly rejected due to backup being disabled")
+}
+
+// testEtcdOpsTaskRejectedIfEtcdNotReady tests that an EtcdOpsTask is rejected if the referenced etcd is not in ready state.
+func testEtcdOpsTaskRejectedIfEtcdNotReady(t *testing.T, namespace string, reconcilerTestEnv ReconcilerTestEnv) {
+	g := NewWithT(t)
+
+	etcdInstance := testutils.EtcdBuilderWithDefaults("test-etcd", namespace).Build()
+	etcdInstance.Status.Conditions = []druidv1alpha1.Condition{
+		{
+			Type:    druidv1alpha1.ConditionTypeReady,
+			Status:  druidv1alpha1.ConditionFalse,
+			Message: "Etcd is not ready for testing",
+		},
+	}
+
+	ctx := context.Background()
+	cl := reconcilerTestEnv.itTestEnv.GetClient()
+
+	g.Expect(cl.Create(ctx, etcdInstance)).To(Succeed())
```

Comment:
Ideally post creation call you should wait till you see the resource. Then you move onto the next create.